### ISSUE TITLE
currentPath is not updated when element position in array changes

### ISSUE THREAD

**Issue by christianliebel** [ElliotNB/observable-slim#15]
When an element’s position in the array changes, this change is not reflected in the `currentPath` and `jsonPointer` properties of the change object:

**Repro**
```js
const ObservableSlim = require('observable-slim');
const proxy = ObservableSlim.create([{}, {}]);

ObservableSlim.observe(proxy, changes => changes
  .forEach(change => console.log(change.currentPath, change.type)));

proxy.splice(0, 1);
proxy[0].foo = 'bar';
```
*Before:* `[{}, {}]`
*After:* `[{foo: 'bar'}]`

**Result**
```
0 update
1 delete
length update
1.foo add
```

However, it should read `0.foo add`, as the object now is the first item of the array.

**Comment by ElliotNB** [comment]
Great catch -- I'll definitely get this scenario added to the unit tests. 

Unfortunately I think the fix for this bug is going to be a bit tricky. Currently the currentPath is tracked recursively using a plain string (`var path` at the top of the `_create` function) but it will need to be converted to an `Array` so that the array positions can be calculated dynamically whenever `_getPath` is invoked. This will also cause a slight knock in performance when we're merging large arrays or objects (making a lot of mutations).

**Comment by christianliebel** [comment]
@ElliotNB Thanks for your reply and the insights. Do you plan to fix this? And if so, do you have a timeline for that?

**Comment by ElliotNB** [comment]
@christianliebel I worked on this issue today and I have it very nearly fixed. I'm hoping to push the fix either tomorrow or Monday.

**Comment by christianliebel** [comment]
@ElliotNB Wow, that’s awesome. Thanks a lot!

**Comment by christianliebel** [comment]
@ElliotNB Looks good, thanks a lot for implementing this fix! I’ve encountered another array-related issue though (at least I think so), but I’ll open another issue for that.

Thank you.
